api/internal/git/cloner.go

```diff
@@ -18,84 +14,30 @@ type Cloner func(repoSpec *RepoSpec) error
 // to say, some remote API, to obtain a local clone of
 // a remote repo.
 func ClonerUsingGitExec(repoSpec *RepoSpec) error {
-	gitProgram, err := exec.LookPath("git")
-	if err != nil {
-		return errors.Wrap(err, "no 'git' program on path")
-	}
-	repoSpec.Dir, err = filesys.NewTmpConfirmedDir()
-	if err != nil {
-		return err
-	}
-
-	cmd := exec.Command(
-		gitProgram,
-		"init",
-		repoSpec.Dir.String())
-	out, err := cmd.CombinedOutput()
-	if err != nil {
-		log.Printf("Error initializing git repo: %s", out)
-		return errors.Wrapf(
-			err,
-			"trouble initializing git repo in %s",
-			repoSpec.Dir.String())
-	}
-
-	cmd = exec.Command(
-		gitProgram,
-		"remote",
-		"add",
-		"origin",
-		repoSpec.CloneSpec())
-	cmd.Dir = repoSpec.Dir.String()
-	out, err = cmd.CombinedOutput()
-	if err != nil {
-		log.Printf("Error adding remote: %s", out)
-		return errors.Wrapf(err, "trouble adding remote %s", repoSpec.CloneSpec())
-	}
-
-	ref := "HEAD"
-	if repoSpec.Ref != "" {
-		ref = repoSpec.Ref
-	}
-
-	cmd = exec.Command(
-		gitProgram,
-		"fetch",
-		"--depth=1",
-		"origin",
-		ref)
-	cmd.Dir = repoSpec.Dir.String()
-	out, err = cmd.CombinedOutput()
-	if err != nil {
-		log.Printf("Error fetching ref: %s", out)
-		return errors.Wrapf(err, "trouble fetching %s", ref)
-	}
-
-	cmd = exec.Command(
-		gitProgram,
-		"checkout",
-		"FETCH_HEAD")
-	cmd.Dir = repoSpec.Dir.String()
-	out, err = cmd.CombinedOutput()
-	if err != nil {
-		log.Printf("Error checking out ref: %s", out)
-		return errors.Wrapf(err, "trouble checking out %s", ref)
-	}
-
-	cmd = exec.Command(
-		gitProgram,
-		"submodule",
-		"update",
-		"--init",
-		"--recursive")
-	cmd.Dir = repoSpec.Dir.String()
-	out, err = cmd.CombinedOutput()
-	if err != nil {
-		log.Printf("Error fetching submodules: %s", out)
-		return errors.Wrapf(err, "trouble fetching submodules for %s", repoSpec.CloneSpec())
-	}
-
-	return nil
+	r, er := newCmdRunner()
+	if er != nil {
+		return er
+	}
+	repoSpec.Dir = r.dir
+	if er = r.run("init"); er != nil {
+		return er
+	}
+	if er = r.run("remote", "add", "origin", repoSpec.CloneSpec()); er != nil {
+		return er
+	}
+	{
```

Comment:
Why this brace?

Reply:
confines the scope of `ref`.  It's such a small method now i can remove it.